kommunicate/src/main/java/com/applozic/mobicomkit/channel/service/ChannelService.java

```diff
@@ -243,7 +243,7 @@ public List<Channel> getChannelList() {
         return channelDatabaseService.getAllChannels();
     }
 
-    public synchronized void syncChannels(boolean isMetadataUpdate) {
+    public void syncChannels(boolean isMetadataUpdate) {
         try {
             final MobiComUserPreference userpref = MobiComUserPreference.getInstance(context);
```

Comment:
Check this one also, it might create issues as we are updating time. Also, this might leads to too many calls in a small amount of time.

Reply:
The issue with this was: Because of this syncChannels() method, other synchronized methods were getting thread locked, because other methods were also synchronized.
Now, I have made this method syncChannels() synchronized, but removed other method's synchronization. This should work because we are sending a Broadcast whenever we find that there is an update in Channel.